[Quant][core][gpu][improvement] Refactored implementation for conv2d_cudnn to use packed parameters (Reland PR#73510)

dzdang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #74227
* __->__ #74220

Summary:
This a reland of https://github.com/pytorch/pytorch/pull/73510 --
please look at this PR directly for a summary and test plan.

The only change in this PR is we add the ops to check_forward_backward_compatibility.py
to get around the backwards compatibility issue introduced in the previous PR that
changes the name of the cudnn conv operations. We also renamed cudnnpack_utils.h
to utils.h.

Differential Revision: [D34886988](https://our.internmc.facebook.com/intern/diff/D34886988)

dzdang committed:
[Quant][core][gpu][improvement] Refactored implementation for conv2d_cudnn to use packed parameters (Reland PR#73510)

Summary:
This a reland of https://github.com/pytorch/pytorch/pull/73510 --
please look at this PR directly for a summary and test plan.

The only change in this PR is we add the ops to check_forward_backward_compatibility.py
to get around the backwards compatibility issue introduced in the previous PR that
changes the name of the cudnn conv operations.

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34886988).

dzdang commented on aten/src/ATen/native/quantized/cudnn/cudnnpack_utils.h:
```diff
@@ -0,0 +1,125 @@
+#pragma once
```
Yeah I can change it. In a later PR (https://github.com/pytorch/pytorch/pull/73959/files), I actually introduce another file `utils.h` file with a `cudnn_utils` namespace, but maybe I can change the naming convention on the later PR instead

jerryzh168 replied:
I see, I think it would make sense to put all utils in one place if possible

dzdang replied:
Do you think it'd be better to just use `utils.h` isntead of `cudnn_utils.h` since everything is already inside the `cudnn` folder?

jerryzh168 replied:
yeah I feel that makes sense

dzdang committed:
Update on "[Quant][core][gpu][improvement] Refactored implementation for conv2d_cudnn to use packed parameters (Reland PR#73510)"


Summary:
This a reland of https://github.com/pytorch/pytorch/pull/73510 --
please look at this PR directly for a summary and test plan.

The only change in this PR is we add the ops to check_forward_backward_compatibility.py
to get around the backwards compatibility issue introduced in the previous PR that
changes the name of the cudnn conv operations. We also renamed cudnnpack_utils.h
to utils.h.

Differential Revision: [D34886988](https://our.internmc.facebook.com/intern/diff/D34886988)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][gpu][improvement] Refactored implementation for conv2d_cudnn to use packed parameters (Reland PR#73510)"


Summary:
This a reland of https://github.com/pytorch/pytorch/pull/73510 --
please look at this PR directly for a summary and test plan.

The only change in this PR is we add the ops to check_forward_backward_compatibility.py
to get around the backwards compatibility issue introduced in the previous PR that
changes the name of the cudnn conv operations. We also renamed cudnnpack_utils.h
to utils.h.

Differential Revision: [D34886988](https://our.internmc.facebook.com/intern/diff/D34886988)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][gpu][improvement] Refactored implementation for conv2d_cudnn to use packed parameters (Reland PR#73510)"


Summary:
This a reland of https://github.com/pytorch/pytorch/pull/73510 --
please look at this PR directly for a summary and test plan.

The only change in this PR is we add the ops to check_forward_backward_compatibility.py
to get around the backwards compatibility issue introduced in the previous PR that
changes the name of the cudnn conv operations. We also renamed cudnnpack_utils.h
to utils.h.

Differential Revision: [D34886988](https://our.internmc.facebook.com/intern/diff/D34886988)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][gpu][improvement] Refactored implementation for conv2d_cudnn to use packed parameters (Reland PR#73510)"


Summary:
This a reland of https://github.com/pytorch/pytorch/pull/73510 --
please look at this PR directly for a summary and test plan.

The only change in this PR is we add the ops to check_forward_backward_compatibility.py
to get around the backwards compatibility issue introduced in the previous PR that
changes the name of the cudnn conv operations. We also renamed cudnnpack_utils.h
to utils.h.

Differential Revision: [D34886988](https://our.internmc.facebook.com/intern/diff/D34886988)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34886988).

jjsjann123 commented:
Did we drop the `cudnn_utils.h` file in the PR when we try to rename `cudnnpack_utils.h`??

jjsjann123 commented:
I have to manually copy this and rename it to `cudnn_utils.h` to build with cudnn v8.
 https://github.com/pytorch/pytorch/blob/406740b3ed34970d8b4c4272845fcc0b8e50822d/aten/src/ATen/native/quantized/cudnn/cudnnpack_utils.h

dzdang commented:
> I have to manually copy this and rename it to `cudnn_utils.h` to build with cudnn v8. https://github.com/pytorch/pytorch/blob/406740b3ed34970d8b4c4272845fcc0b8e50822d/aten/src/ATen/native/quantized/cudnn/cudnnpack_utils.h

Yes, we renamed it `cudnnpack_utils.h` to `cudnn_utils.h` for now. I think we're going to change some more of the file names in a future PR to have it be more aligned with the files in the non-quantized directories. Are you working with a previous version?

dzdang commented:
> Did we drop the `cudnn_utils.h` file in the PR when we try to rename `cudnnpack_utils.h`??

@jjsjann123
oh wait, yes, I see now. I need to make this change

xwang233 commented:
@malfet This PR is causing our cudnn v8 build to fail. One of the included files doesn't exist.

```cpp
#include <ATen/native/quantized/cudnn/cudnn_utils.h>
```

dzdang commented:
> @malfet This PR is causing our cudnn v8 build to fail. One of the included files doesn't exist.
> 
> ```c++
> #include <ATen/native/quantized/cudnn/cudnn_utils.h>
> ```

@xwang223 @malfet 
We are going to revert this PR or add a patch. This file: https://github.com/pytorch/pytorch/blob/406740b3ed34970d8b4c4272845fcc0b8e50822d/aten/src/ATen/native/quantized/cudnn/cudnnpack_utils.h was renamed to `cudnn_utils.h` but was accidentally dropped from the amended commit. If you need a quick fix you can use this file and rename it `cudnn_utils.h`

dzdang commented:
@xwang233 @jjsjann123 I reverted this PR. Master should be fine now when building with cudnn v8
